bundles/org.openhab.binding.neohub/README.md

```diff
@@ -33,21 +33,31 @@ It signs on to the hub using the supplied connection parameters, and it polls th
 The NeoHub supports two Application Programming Interfaces "API" (an older "legacy" one, and a modern one), and this binding can use either of them to communicate with it.
 Before the binding can communicate with the hub, the following Configuration Parameters must be entered.
 
-| Configuration Parameter | Description                                                                                 |
-|-------------------------|---------------------------------------------------------------------------------------------|
-| hostName                | Host name (IP address) of the NeoHub (example 192.168.1.123)                                |
-| portNumber              | Port number of the NeoHub (Default=4242)                                                    |
-| pollingInterval         | Time (seconds) between polling requests to the NeoHub (Min=4, Max=60, Default=60)           |
-| socketTimeout           | Time (seconds) to allow for TCP socket connections to the hub to succeed (Min=4, Max=20, Default=5) |
-| preferLegacyApi         | ADVANCED: Prefer the binding to use older API calls; if these are not supported, it switches to the new calls (Default=false) |
+| Configuration Parameter    | Description                                                                                              |
+|----------------------------|----------------------------------------------------------------------------------------------------------|
+| hostName                   | Host name (IP address) of the NeoHub (example 192.168.1.123)                                             |
+| useWebSockets<sup>1)</sup> | Use secure WebSockets to connect to  the NeoHub (example `true`)                                         |
+| apiToken<sup>1)</sup>      | API Access Token for secure connection to hub. Create the token in the Heatmiser mobile App              |
+| pollingInterval            | Time (seconds) between polling requests to the NeoHub (Min=4, Max=60, Default=60)                        |
+| socketTimeout              | Time (seconds) to allow for TCP socket connections to the hub to succeed (Min=4, Max=20, Default=5)      |
+| preferLegacyApi            | ADVANCED: Prefer to use older API calls; but if not supported, it switches to new calls (Default=false)  |
```

Comment:
Forgive me for one more question, perhaps even a stupid one: Can the legacy API be used with web sockets, or are they mutually exclusive? I'm thinking if protocol selection could be something like "Legacy", "Insecure" (?) and "Web Sockets", unless of course all combinations of the two boolean parameters are supported. The other thing which is not clear to me from reading this README - below you describe the old protocol as "older and less secure". That makes me think only about encryption. So just to be clear: The old protocol keeps a TCP connection open and also receives push messages this way? So in terms of functionality they are kind of similar, but the web socket implementation uses HTTPS, thus the major benefit is security by encryption?

Reply:
See this as background..

https://dev.heatmiser.com/t/api-neohub-v3-0-protocol-8th-june-2022/222?u=andrewfg

The words new/legacy apply (confusingly) to two different things; namely the application layer and the transport layer.
1. Application layer. Depending on the firmware, a hub can support two dialects of JSON request and response payloads. The binding supports both, and it can auto detect and auto select which to use. In the case that the hub supports both dialects, the ‘preferLegacyApi’ config param allows you to decide which dialect shall be used.
2. Transport layer. Depending on the firmware, and a setting in the app, a hub can support two transport layers. Namely a TCP connection and/or a Websocket connection. The TCP connection is plaintext, and neither encrypted nor authenticated. The web socket connection is both encrypted and authenticated via a user generated access key.
3. In principle a hub can support both types of application layer, and both types of transport layer. So four possible permutations.

PS neither of the application layers supports push notifications. Both are query / response models.

Reply:
Thanks. In that case the current solution seems optimal.